processor/src/main/java/org/derive4j/processor/derivator/LazyConstructorDerivator.java

```diff
@@ -73,20 +73,20 @@
                 .addModifiers(Modifier.PRIVATE)
                 .returns(typeName)
                 .addCode(CodeBlock.builder()
-                    .addStatement("$T _evaluation = this.evaluation", typeName)
-                    .beginControlFlow("if (_evaluation == null)")
-                    .beginControlFlow("synchronized (this.lock)")
-                    .addStatement("_evaluation = this.evaluation")
-                    .beginControlFlow("if (_evaluation == null)")
-                    .addStatement("this.evaluation = _evaluation = expression.$L()", Utils.getAbstractMethods(lazyTypeElement.getEnclosedElements()).get(0).getSimpleName())
-                    .addStatement("this.expression = null")
```

Comment:
the expression was set to null also to free-up memory. with your patch this will no longer be the case.